data-grabber.py

```diff
@@ -0,0 +1,106 @@
+welshBaseRequestUrl = "http://agored.statscymru.llyw.cymru/"
+outputfile = "out.csv"
+datasetID = "XXXX0000".lower()
+
+def get_next_link(url):
+    print(f"Getting data from '{url}'")
+    try:
+        r = requests.get(url)
+    except RuntimeError as error:
+        print(error)
+        print(f"Unable to make connection to petitions url '{url}'. Exiting...")
+        quit()
+    list_data = r.json()
+    return list_data
+
+def createdatafile(outfile, data):
+    with open(outfile, 'w', newline='') as csvfile:
+        csvwriter = csv.writer(csvfile, delimiter=',',
+                                quotechar='"', quoting=csv.QUOTE_MINIMAL)
+        topRow = []
+        for key in data[0].keys():
+            topRow.append(key)
+        csvwriter.writerow(topRow)
+        csvfile.close()
+        return topRow
+
+def writedata(outfile, toprow, data):
+         with open(outfile, 'a', newline='') as csvfile:
+            csvwriter = csv.writer(csvfile, delimiter=',',
+                                quotechar='"', quoting=csv.QUOTE_MINIMAL)
+            for item in data:
+                row = []
+                for key in toprow:
+                    row.append(item[key])
+                csvwriter.writerow(row)
+            csvfile.close()
+
+def main(argv):
+    global welshBaseRequestUrl, englistBaseRequestUrl
+    parser=argparse.ArgumentParser(description="Get data from statswales.gov.wales and save to a csv file.")
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument("-u", "--url", help="a StatsWales open data url", type=str)
+    group.add_argument("-d", "--dataset", help="A StatsWales opendata name (You can find this on the Metadata tab)", type=str)
+    parser.add_argument("-w", "--welsh", help="Force getting data from the welsh languague version", action="store_true")
+    parser.add_argument("-e", "--english", help="Force getting data from the english languague version", action="store_true")
+    parser.add_argument("-o", "--out", help="Override the name of the output file (default: the name of the dataset)", type=str)
+    args = parser.parse_args()
+
+    if args.dataset:
+        datasetID = args.dataset.lower()
+        englistInitalRequestUrl = f"{englistBaseRequestUrl}/{datasetID}"
+        welshInitalRequestUrl = f"{welshBaseRequestUrl}/{datasetID}"
+        if args.welsh:
+            initalRequestUrl = welshInitalRequestUrl
+        else:
+            initalRequestUrl = englistInitalRequestUrl
+    elif args.url:
+        if welshBaseRequestUrl in args.url:
+            welshInitalRequestUrl = args.url
+            welsh = True
+        elif englistBaseRequestUrl in args.url:
+            englistInitalRequestUrl = args.url
+            welsh = False
+        else:
+            print("Invalid URL. You need to supply a URL from StatsWales. Exiting...")
+            exit(1)
+        initalRequestUrl = args.url
+        datasetID = initalRequestUrl.split("/")[-1].lower()
```

Comment:
Is there an API to mutate the paths in the URL? It's be less open to injection attacks/mistakes in the received JSON.

i.e. Using this string mutation method, if any of the path components contain '/' they will end up in the URL verbatim rather than being escaped. A URL library should handle that for us.

Reply:
I can switch to using Pythons URL library